verify_stream.go

```diff
@@ -83,36 +118,60 @@ func (v *verifyStream) readHeader(versionValidator VersionValidator, msgType Mes
 	if err := header.validate(versionValidator, msgType); err != nil {
 		return err
 	}
+	v.version = header.Version
 	v.state = stateBody
 	return nil
 }
 
-func (v *verifyStream) readBlock(p []byte) (int, bool, error) {
-	var block signatureBlock
-	_, err := v.stream.Read(&block)
+func readSignatureBlock(version Version, mps *msgpackStream) (signature, payloadChunk []byte, isFinal bool, seqno packetSeqno, err error) {
+	defer func() {
+		if err == nil {
+			// The header packet picks up the zero seqno,
+			// so subtract 1 to compensate for that.
+			seqno--
```

Comment:
Ah, looking in the encryption/signcryption code, I see where it's decrementing by 1. I'll see if I can make it consistent.